src/components/NodeEditor/NodeEditor.tsx

```diff
@@ -78,16 +83,95 @@ export interface NodeEditorProps {
     onUpdateLocalizations: Function;
     onUpdateAction: Function;
     onUpdateRouter: Function;
-    /** Perform when editor is closed */
     onClose?(canceled: boolean): void;
     ComponentMap: ComponentMap;
 }
 
 export interface NodeEditorState {
     config: Type;
     show: boolean;
+    setResultName: boolean;
 }
 
+export const mapExits = (exits: Exit[]): { [uuid: string]: Exit } =>
+    exits.reduce(
+        (map, exit) => {
+            map[exit.uuid] = exit;
+            return map;
+        },
+        {} as { [uuid: string]: Exit }
+    );
+
+export const isSwitchForm = (type: string) =>
+    type === 'wait_for_response' || type === 'expression' || type === 'group';
+
+/**
+ * Returns existing action (if any), or a bare-bones representation of the form's action.
+ */
+export const getAction = ({ action }: NodeEditorProps, config: Type): AnyAction => {
```

Comment:
Moving this to a function expression so it's easier to test/not bloating the NodeEditor component. Otherwise, in our tests, we'd need to instantiate `NodeEditor` before testing this method.